### PR TITLE
Feature/et 1342 stripe payments

### PR DESCRIPTION
**Pull request by moraleida**
### 🎫 Ticket

[ET-1342] 

### 🎥 Artifacts 
🎥 https://www.loom.com/share/6d12f0ad539848a4818273ce9878693c

[ET-1342]: https://theeventscalendar.atlassian.net/browse/ET-1342?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/Tickets/Commerce/Gateways/Stripe/Hooks.php [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+	public function handle_action_connected() {
+
+		if ( empty( $_GET['stripe'] ) ) {
+			return;
+		}
+
+		tribe( Signup::class )->handle_connection_established();
+	}
+
+	public function handle_stripe_errors() {
```
Comment (bordoni) [inline src/Tickets/Commerce/Gateways/Stripe/Hooks.php:75]: Docblock.

File: src/Tickets/Commerce/Gateways/Stripe/Hooks.php [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+
+		if ( empty( $_GET['tc-stripe-error'] ) ) {
+			return;
+		}
+
+		tribe( Notice_Handler::class )->trigger_admin( $_GET[ 'tc-stripe-error' ] );
```
Comment (bordoni) [inline src/Tickets/Commerce/Gateways/Stripe/Hooks.php:81]: Use `tribe_get_request_var`

File: src/Tickets/Commerce/Gateways/Stripe/Hooks.php [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+	}
+
+	/**
+	 * Receive data after a connection to stripe has been established
+	 *
+	 * @since TBD
+	 */
+	public function handle_action_connected() {
+
+		if ( empty( $_GET['stripe'] ) ) {
```
Comment (bordoni) [inline src/Tickets/Commerce/Gateways/Stripe/Hooks.php:68]: Use `tribe_get_request_var`

File: src/Tickets/Commerce/Gateways/Stripe/Hooks.php [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+	 * @inheritDoc
+	 */
+	public function register() {
+		$this->add_actions();
+		$this->add_filters();
+	}
+
+	/**
+	 * @since TBD
+	 * @return mixed
```
Comment (bordoni) [inline src/Tickets/Commerce/Gateways/Stripe/Hooks.php:19]: Docblock missing part and wrong return.

File: src/Tickets/Commerce/Gateways/Stripe/Hooks.php [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+	 * Include the payment element from Stripe into the Checkout page.
+	 *
+	 * @since TBD
+	 *
+	 * @param string           $file     Which file we are loading.
+	 * @param string           $name     Name of file file
+	 * @param \Tribe__Template $template Which Template object is being used.
+	 */
+	public function include_payment_buttons( $file, $name, $template ) {
+		$this->container->make( PaymentElement::class )->include_payment_element( $file, $name, $template );
```
Comment (bordoni) [inline src/Tickets/Commerce/Gateways/Stripe/Hooks.php:108]: PaymentElement here has the wrong Standard for Casing.

File: src/Tickets/Commerce/Gateways/Stripe/PaymentElement.php [hunk 6]
```diff
@@ -0,0 +1,24 @@
+<?php
+
+namespace TEC\Tickets\Commerce\Gateways\Stripe;
+
+class PaymentElement {
```
Comment (bordoni) [inline src/Tickets/Commerce/Gateways/Stripe/PaymentElement.php:5]: Wrong casing for the Payment Element class.

File: src/Tickets/Commerce/Gateways/Stripe/REST.php [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+	/**
+	 * Register the endpoints for handling webhooks.
+	 *
+	 * @since 5.1.6
+	 */
+	public function register_endpoints() {
+		$this->container->make( REST\Publishable_Key_Endpoint::class )->register();
+		$this->container->make( REST\Order_Endpoint::class )->register();
+	}
+}
```
Comment (bordoni) [inline src/Tickets/Commerce/Gateways/Stripe/REST.php:28]: Wrong Since here.

File: src/Tickets/Commerce/Gateways/Stripe/Provider.php [hunk 8]
```diff
@@ -0,0 +1,41 @@
+<?php
+
+namespace TEC\Tickets\Commerce\Gateways\Stripe;
+
+class Provider extends \tad_DI52_ServiceProvider {
```
Comment (bordoni) [inline src/Tickets/Commerce/Gateways/Stripe/Provider.php:5]: Missing Docblock.

File: src/Tickets/Commerce/Gateways/Stripe/Merchant.php [hunk 9]
```diff
@@ -0,0 +1,99 @@
...
+	 *
+	 * @return string
+	 */
+	public function get_signup_data_key() {
+		$gateway_key   = Gateway::get_key();
+
+		return "tec_tickets_commerce_{$gateway_key}_signup_data";
+	}
+
+	public function get_client_secret() {
```
Comment (bordoni) [inline src/Tickets/Commerce/Gateways/Stripe/Merchant.php:64]: Missing docblock.

File: src/Tickets/Commerce/Gateways/Stripe/Merchant.php [hunk 10]
```diff
@@ -0,0 +1,99 @@
+<?php
+
+namespace TEC\Tickets\Commerce\Gateways\Stripe;
+
+use TEC\Tickets\Commerce\Gateways\Contracts\Abstract_Merchant;
+
+class Merchant extends Abstract_Merchant {
```
Comment (bordoni) [inline src/Tickets/Commerce/Gateways/Stripe/Merchant.php:7]: Missing Docblokc.

File: src/Tickets/Commerce/Gateways/Stripe/Merchant.php [hunk 11]
```diff
@@ -0,0 +1,99 @@
...
+
+	protected $client_id;
+
+	protected $refresh_tokens;
+
```
Comment (bordoni) [inline src/Tickets/Commerce/Gateways/Stripe/Merchant.php:12]: Missing these 2 docblocks.

File: src/Tickets/Commerce/Gateways/Stripe/Signup.php [hunk 12]
```diff
@@ -0,0 +1,101 @@
...
+		} else {
+			$data->{'tc-stripe-signup-error'} = true;
+			$this->signup_return_path = add_query_arg( (array) $data, $this->signup_return_path );
+		}
+
+		wp_safe_redirect( admin_url( $this->signup_return_path ) );
+		exit();
+	}
+
+	public function is_success( $data ) {
```
Comment (bordoni) [inline src/Tickets/Commerce/Gateways/Stripe/Signup.php:95]: Missing docblock.

File: src/Tickets/Commerce/Gateways/Stripe/Signup.php [hunk 13]
```diff
@@ -0,0 +1,101 @@
+<?php
+
+namespace TEC\Tickets\Commerce\Gateways\Stripe;
+
+use TEC\Tickets\Commerce\Gateways\Contracts\Abstract_Signup;
+use TEC\Tickets\Commerce\Gateways\Stripe\REST\On_Boarding_Endpoint;
+
+class Signup extends Abstract_Signup {
```
Comment (bordoni) [inline src/Tickets/Commerce/Gateways/Stripe/Signup.php:8]: Missing docblock

File: src/Tickets/Commerce/Gateways/Stripe/WhoDat.php [hunk 14]
```diff
@@ -0,0 +1,100 @@
...
+
+	/**
+	 * The API URL.
+	 *
+	 * @since TBD
+	 *
+	 * @var string
+	 */
+//	protected $api_url = 'https://whodat.theeventscalendar.com/commerce/v1/stripe';
+	protected $api_url = 'https://localhost/whodat/index.php/commerce/v1/stripe';
```
Comment (bordoni) [inline src/Tickets/Commerce/Gateways/Stripe/WhoDat.php:25]: We should modify this on the Abstract so we can change with a variable on your WP Config.

File: src/admin-views/settings/tickets-commerce/stripe/connect/active.php [hunk 15]
```diff
@@ -0,0 +1,35 @@
+<?php
+/**
+ * The Template for displaying the Tickets Commerce PayPal Settings when connected.
+ *
+ * @version 5.1.10
+ *
+ * @since   5.1.10
```
Comment (bordoni) [inline src/admin-views/settings/tickets-commerce/stripe/connect/active.php:7]: Since and version are wrong, and docblock is mentioning paypal still.

File: src/admin-views/settings/tickets-commerce/stripe/connect/active.php [hunk 16]
```diff
@@ -0,0 +1,35 @@
...
+ *
+ * @var Tribe__Tickets__Admin__Views                  $this                  [Global] Template object.
+ * @var string                                        $plugin_url            [Global] The plugin URL.
+ * @var TEC\Tickets\Commerce\Gateways\PayPal\Merchant $merchant              [Global] The merchant class.
+ * @var TEC\Tickets\Commerce\Gateways\PayPal\Signup   $signup                [Global] The Signup class.
```
Comment (bordoni) [inline src/admin-views/settings/tickets-commerce/stripe/connect/active.php:12]: PayPal to Stripe.

File: src/admin-views/settings/tickets-commerce/stripe/connect/active/actions.php [hunk 17]
```diff
@@ -0,0 +1,32 @@
...
+ * The Template for displaying the Tickets Commerce PayPal connection details.
+ *
+ * @version 5.2.0
+ *
+ * @since 5.2.0
+ *
+ * @var Tribe__Tickets__Admin__Views                  $this               [Global] Template object.
+ * @var string                                        $plugin_url         [Global] The plugin URL.
+ * @var TEC\Tickets\Commerce\Gateways\PayPal\Merchant $merchant           [Global] The merchant class.
+ * @var TEC\Tickets\Commerce\Gateways\PayPal\Signup   $signup             [Global] The Signup class.
```
Comment (bordoni) [inline src/admin-views/settings/tickets-commerce/stripe/connect/active/actions.php:12]: Same as the other template.

File: src/admin-views/settings/tickets-commerce/stripe/connect/active/actions/refresh-access-token.php [hunk 18]
```diff
@@ -0,0 +1,28 @@
...
+ * The Template for displaying the Tickets Commerce refresh access token action button.
+ *
+ * @version 5.2.0
+ *
+ * @since 5.2.0
+ *
+ * @var Tribe__Tickets__Admin__Views                  $this               [Global] Template object.
+ * @var string                                        $plugin_url         [Global] The plugin URL.
+ * @var TEC\Tickets\Commerce\Gateways\PayPal\Merchant $merchant           [Global] The merchant class.
+ * @var TEC\Tickets\Commerce\Gateways\PayPal\Signup   $signup             [Global] The Signup class.
```
Comment (bordoni) [inline src/admin-views/settings/tickets-commerce/stripe/connect/active/actions/refresh-access-token.php:12]: Incorrect PayPal, should be Stripe.

File: src/resources/js/commerce/gateway/stripe/checkout.js [hunk 19]
```diff
@@ -0,0 +1,152 @@
...
+		};
+
+		// Handle submit
+		var paymentButton = document.getElementById( obj.selectors.button );
+		paymentButton.addEventListener( 'click', obj.callbacks.submit );
+	};
+
+	obj.bindEvents();
+
+})( jQuery, tribe.tickets.commerce.gateway.stripe, Stripe );
```
Comment (bordoni) [inline src/resources/js/commerce/gateway/stripe/checkout.js:152]: We need to move a bunch of things here to use ES6 standard.

File: src/views/v2/commerce/gateway/stripe/payment_element.php [hunk 20]
```diff
@@ -0,0 +1,31 @@
...
+<form id="payment-form">
+	<div id="card-element">
+		<!-- Elements will create input elements here -->
+	</div>
+
+	<!-- We'll put the error messages in this element -->
+	<div id="card-errors" role="alert"></div>
+
+	<button id="tec-tc-gateway-stripe-checkout-button">Submit Payment</button>
+</form>
```
Comment (bordoni) [inline src/views/v2/commerce/gateway/stripe/payment_element.php:31]: Is there any way we can change the IDs here? Just to make sure we are namespacing it.

**Review by bordoni — CHANGES_REQUESTED** [review]
All the file templates should be using `@since` and `@version` with TBD.

This feels like a solid update to the codebase, my comments are mostly around Code standards and docblocks.

I might have more to say about the JS but I will wait until later this week to tackle it.

File: src/views/v2/commerce/gateway/stripe/payment_element.php (hunk 20 above)
Comment (moraleida) [reply src/views/v2/commerce/gateway/stripe/payment_element.php]: @bordoni I looked around the documentation and couldn't find a way to change the `payment-form` ID, but the others are taken care of.

File: src/Tickets/Commerce/Gateways/Contracts/Abstract_WhoDat.php [hunk 21]
```diff
@@ -13,11 +13,47 @@
...
+	 * Returns the WhoDat URL to use.
+	 *
+	 * @since TBD
+	 *
+	 * @return string
+	 */
+	private function get_api_base_url() {
+
+		if ( defined( 'TEC_TC_WHODAT_DEV_URL' ) && TEC_TC_WHODAT_DEV_URL ) {
+			return TEC_TC_WHODAT_DEV_URL;
```
Comment (borkweb) [inline src/Tickets/Commerce/Gateways/Contracts/Abstract_WhoDat.php:46]: Let's change this to `TEC_TC_API_PROXY_URL`.

File: src/Tickets/Commerce/Gateways/Stripe/Client.php [hunk 22]
```diff
@@ -0,0 +1,277 @@
...
+	 *
+	 * @return string
+	 */
+	public function get_environment_url() {
+		$merchant = tribe( Merchant::class );
+
+		// @todo determine if the sandbox of stripe has a diff URL.
+		return $merchant->is_sandbox() ?
+			'https://api.stripe.com/v1' :
+			'https://api.stripe.com/v1';
```
Comment (borkweb) [inline src/Tickets/Commerce/Gateways/Stripe/Client.php:28]: Make this filterable.

File: src/Tickets/Commerce/Gateways/Stripe/Merchant.php [hunk 23]
```diff
@@ -47,8 +111,9 @@ public function get_signup_data_key() {
 	 */
 	public function to_array() {
 		return [
-			'client_id'      => $this->get_client_id(),
-			'refresh_tokens' => $this->get_refresh_tokens(),
+			'client_id'       => $this->get_client_id(),
+			'client_secret'   => $this->get_client_secret(),
+			'publishable_key' => $this->get_client_id(),
```
Comment (borkweb) [inline src/Tickets/Commerce/Gateways/Stripe/Merchant.php:116]: I think this should be:

```php
'publishable_key' => $this->get_publishable_key(),
```

File: src/Tickets/Commerce/Gateways/Stripe/REST/Publishable_Key_Endpoint.php [hunk 24]
```diff
@@ -52,32 +45,38 @@ public function register() {
...
+	 */
+	public function get_key( WP_REST_Request $request ) {
+
+		$params = $request->get_json_params();
+
+		/* @todo fixme
+		if ( ! wp_verify_nonce( $params['nonce'], Assets::PUBLISHABLE_KEY_NONCE_ACTION ) ) {
+			wp_send_json_error( 'Invalid nonce ' . $params['nonce'] );
+		}
+		*/
```
Comment (borkweb) [inline src/Tickets/Commerce/Gateways/Stripe/REST/Publishable_Key_Endpoint.php:75]: Flagging this, though I imagine this is going to be fixed soon in another PR.
